src/Illuminate/Validation/Rules/Date.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Illuminate\Validation\Rules;
+
+use DateTime;
+use DateTimeImmutable;
+
+class Date
+{
+    /** @var string $type The type of the rule to apply. */
+    protected $type;
+
+    /** @var string|DateTime $date The date or field name to compare the type against. */
+    protected $date;
+
+    /** @var array allowedTypes The allowed types of the rule. */
+    const allowedTypes = [
+        'date',
+        'after',
+        'after_or_equal',
+        'before',
+        'before_or_equal',
+    ];
+
+    /**
+     * Create a new date rule instance.
+     *
+     * @param string $type
+     * @param string|DateTime $date
+     * @return void
+     */
+    public function __construct($type = 'date', $date = '')
+    {
+        $this->type($type);
+        $this->date($date);
+    }
+
+    /**
+     * Set the type of the rule manually.
+     *
+     * @param string $value
+     * @return $this
+     */
+    public function type($value)
+    {
+        if (! in_array($value, self::allowedTypes)) {
+            throw new \InvalidArgumentException();
+        }
+
+        $this->type = $value;
+
+        return $this;
+    }
+
+    /**
+     * Set the date or field to compare the submitted value against.
+     *
+     * @param string|DateTime $value
```

Comment:
Please use FQCNs and a leading slash. (NB fully qualified is defined without a slash, the slash is only used for escaping to the global namespace)